pkg/kubelet/kubelet.go

```diff
@@ -1329,21 +1329,30 @@ func (kl *Kubelet) computePodContainerChanges(pod *api.Pod, hasMirrorPod bool, r
 	}, nil
 }
 
-func (kl *Kubelet) syncPod(pod *api.Pod, hasMirrorPod bool, runningPod kubecontainer.Pod) error {
+func (kl *Kubelet) syncPod(pod *api.Pod, mirrorPod *api.Pod, runningPod kubecontainer.Pod) error {
 	podFullName := kubecontainer.GetPodFullName(pod)
 	uid := pod.UID
 
 	// Before returning, regenerate status and store it in the cache.
 	defer func() {
+		if isStaticPod(pod) && mirrorPod == nil {
+			// No need to cache the status because the mirror pod does not
+			// exist yet.
+			return
+		}
 		status, err := kl.generatePodStatusByPod(pod)
 		if err != nil {
 			glog.Errorf("Unable to generate status for pod with name %q and uid %q info with error(%v)", podFullName, uid, err)
 		} else {
-			kl.statusManager.SetPodStatus(podFullName, status)
+			podToUpdate := pod
+			if mirrorPod != nil {
```

Comment:
Yes, my mistake.

I think we don't test the pod status syncing properly. All the integration tests get pod status from the apiserver, and apiserver queries kubelet if the status does not exist. I guess this would be improved after we remove the pod cache...